### PR TITLE
BCH integration 

### PR DESCRIPTION
**Pull request by badr-r**

### CODE REVIEW

File: php/WC_Gateway_Blockonomics.php [hunk 1]
```diff
@@ -245,8 +263,9 @@ public function check_blockonomics_callback()
                 $order['status'] = $status;
                 $orders[$addr] = $order;
                 if ($existing_status == -1) {
-                    update_post_meta($wc_order->get_id(), 'blockonomics_txid', $order['txid']);
-                    update_post_meta($wc_order->get_id(), 'expected_btc_amount', $order['satoshi']/1.0e8);
+                    update_post_meta($wc_order->get_id(), 'blockonomics_'. $order['crypto'] .'_txid', $order['txid']);
+                    update_post_meta($wc_order->get_id(), 'expected_'. $order['crypto'] .'_amount', $order['satoshi']/1.0e8);
+                    update_post_meta($wc_order->get_id(), ''. $order['crypto'] .'_address', $order['address']);
```
Comment (DarrenWestwood) [inline php/WC_Gateway_Blockonomics.php:268]: The crypto address should be added to the post meta when creating the address.

File: templates/blockonomics_checkout.php [hunk 2]
```diff
@@ -120,6 +122,12 @@
         </div>
       </div>
     </div>
+    <!-- BCH address Generation Error -->
+    <div class="bnomics-display-error" style="text-align: center;" ng-show="bchaddresserror">
+            <h4>Address Generation Error</h4>
+            <h4><i class="material-icons bnomics-alt-icon">error</i></h4>
+            <p>Note to webmaster: Please follow the instructions <a href="https://help.blockonomics.co/support/solutions/articles/33000251576-bch-setup-on-whmcs" target="_blank">here</a> to configure BCH payments.</p>
```
Comment (DarrenWestwood) [inline templates/blockonomics_checkout.php:129]: Incorrect href. Please change this to https://help.blockonomics.co/support/solutions/articles/33000253348-bch-setup-on-woocommerce. This freshdesk article will be updated accordingly.

File: php/WC_Gateway_Blockonomics.php [hunk 3]
```diff
@@ -184,52 +162,92 @@ public function check_blockonomics_callback()
...
+            $order = $orders[$order_id];
+            $wc_order = new WC_Order($order_id);
+            if(get_woocommerce_currency() != 'BTC'){
+                $price = $blockonomics->get_price(get_woocommerce_currency(), $crypto);
+                $price = $price * 100/(100+get_option('blockonomics_margin', 0));
+            }else{
+                $price = 1;
+            }
+            $currentAddress = get_post_meta($order_id,"blockonomics_address");
+            if($currentAddress[0]['address'] && $currentAddress[0]['crypto'] == $crypto) {
```
Comment (DarrenWestwood) [inline php/WC_Gateway_Blockonomics.php:188]: `btc_address` and `bch_address` are added to the post for admin. `blockonomics_address` can be removed and check for existing address can use these `btc_address` and `bch_address` post meta.

File: php/WC_Gateway_Blockonomics.php [hunk 4]
```diff
@@ -108,42 +106,22 @@ public function process_payment($order_id)
...
-        }
-
         $order = array(
             'value'              => $order->get_total(),
-            'satoshi'            => intval(round(1.0e8*$order->get_total()/$price)),
+            'satoshi'            => '',
             'currency'           => get_woocommerce_currency(),
             'order_id'           => $order_id,
+            'crypto'             => 'BTC',
+            'address'            => '',
```
Comment (shivaenigma) [inline php/WC_Gateway_Blockonomics.php:115]: Empty address and crypto is written to orders array. Not a good design ?

**Comment by DarrenWestwood** [comment]
Tested different use cases. The plugin functions as desired for both btc and bch.

One last issue and some code cleanup:
- The qr code and url does not change from `bitcoin:` to `bitcoincash:`
https://github.com/badr-r/woocommerce-plugin/blob/master/templates/blockonomics_checkout.php#L67-L68
- Unused track.php template file can be removed
- Unused font files can be removed. ie cryptofont
- WpAjax not used and can be removed: https://github.com/badr-r/woocommerce-plugin/blob/master/js/app.js#L18
https://github.com/badr-r/woocommerce-plugin/blob/master/blockonomics-woocommerce.php#L387-L429

Side note: The Payment Page UI can better match the design.

**Comment by DarrenWestwood** [comment]
- bitcoincash QR code shows unsafe in the browser. eg. `unsafe:bitcoincash:qrak425ur9z8s06rvxg9jppma78fuyvffy5sjrs70v?amount=0.11690437`.  aHrefSanitizationWhitelist needs to be updated to solve this: https://github.com/badr-r/woocommerce-plugin/blob/master/js/app.js#L22
- There are still remaining cryptofont files, which are not being used. Also check if icon.css is also still being used - 
https://github.com/badr-r/woocommerce-plugin/tree/master/css
https://github.com/badr-r/woocommerce-plugin/blob/master/blockonomics-woocommerce.php#L361-L362

- Payment Page Design 
Bitcoin Payment   Company Name(1)

- Current Pull Request 
Wordpress(14)

* Update UI to fully match the design.
	** Update click here to go back to refresh rather than going back.
	** Amount in input box. Clicking should copy the amount.
	** To pay messages require updating.
	** There is also unneccessary margin around the address input box. This shortens the input field.

**Pull request merged**